lib/client/config_e2e_test.go

```diff
@@ -89,6 +89,89 @@ var _ = testutils.E2eDatastoreDescribe("with config option API tests", testutils
 		Expect(cs).To(Equal(client.ConfigLocationGlobal))
 	})
 
+	It("should handle node IP in IP tunnel address", func() {
+		var err error
+		var ip *net.IP
+
+		By("checking unset value")
+		ip, err = config.GetNodeIPIPTunnelAddress("node1")
+		Expect(err).NotTo(HaveOccurred())
+		Expect(ip).To(BeNil())
+
+		By("checking address set to 1.2.3.4")
+		ipv4 := net.MustParseIP("1.2.3.4")
+		err = config.SetNodeIPIPTunnelAddress("node1", &ipv4)
+		Expect(err).NotTo(HaveOccurred())
+
+		ip, err = config.GetNodeIPIPTunnelAddress("node1")
+		Expect(err).NotTo(HaveOccurred())
+		Expect(*ip).To(Equal(ipv4))
+
+		By("checking address set to aa::ff")
+		ipv6 := net.MustParseIP("aa::ff")
+		err = config.SetNodeIPIPTunnelAddress("node1", &ipv6)
+		Expect(err).NotTo(HaveOccurred())
+
+		ip, err = config.GetNodeIPIPTunnelAddress("node1")
+		Expect(err).NotTo(HaveOccurred())
+		Expect(*ip).To(Equal(ipv6))
+
+		By("checking address set to nil")
+		err = config.SetNodeIPIPTunnelAddress("node1", nil)
+		Expect(err).NotTo(HaveOccurred())
+
+		ip, err = config.GetNodeIPIPTunnelAddress("node1")
+		Expect(err).NotTo(HaveOccurred())
+		Expect(ip).To(BeNil())
+	})
+
+	It("should handle per-node felix config", func() {
```

Comment:
Nope - as with the above, I haven't added per-node Felix config to KDD.  It isn't required for KDD routing.

I think we should add it soon, and then we should be feature complete (in terms of what can be configured through calicoctl).